share-through-htb.js

```diff
@@ -215,142 +183,79 @@ function ShareThroughHtb(configs) {
-            var curBid;
-
-            for (var i = 0; i < bids.length; i++) {
-
-                /**
-                 * This section maps internal returnParcels and demand returned from the bid request.
-                 * In order to match them correctly, they must be matched via some criteria. This
-                 * is usually some sort of placements or inventory codes. Please replace the someCriteria
-                 * key to a key that represents the placement in the configuration and in the bid responses.
-                 */
-
-                if (curReturnParcel.xSlotRef.someCriteria === bids[i].someCriteria) {
-                    curBid = bids[i];
-                    break;
-                }
-            }
-
-            /* ------------------------------------------------------------------------------------*/
-
-            /* HeaderStats information */
-            var headerStatsInfo = {};
-            headerStatsInfo[curReturnParcel.htSlot.getId()] = [curReturnParcel.xSlotName];
-
-            /* No matching bid found so its a pass */
-            if (!curBid) {
-                if (__profile.enabledAnalytics.requestTime) {
-                    __baseClass._emitStatsEvent(sessionId, 'hs_slot_pass', headerStatsInfo);
-                }
-                curReturnParcel.pass = true;
-                continue;
-            }
-
-            /* ---------- Fill the bid variables with data from the bid response here. ------------*/
-            /* Using the above variable, curBid, extract various information about the bid and assign it to
-            * these local variables */
-
-            var bidPrice = curBid.price; /* the bid price for the given slot */
-            var bidSize = [curBid.width, curBid.height]; /* the size of the given slot */
-            var bidCreative = curBid.adm; /* the creative/adm for the given slot that will be rendered if is the winner. */
-            var bidDealId = curBid.dealid; /* the dealId if applicable for this slot. */
-
-            /* ---------------------------------------------------------------------------------------*/
-
+        /* No matching bid found so its a pass */
+        if (curBid.creatives.length === 0) {
             if (__profile.enabledAnalytics.requestTime) {
-                __baseClass._emitStatsEvent(sessionId, 'hs_slot_bid', headerStatsInfo);
+                __baseClass._emitStatsEvent(sessionId, 'hs_slot_pass', headerStatsInfo);
             }
+            curReturnParcel.pass = true;
+            return;
+        }
 
-            curReturnParcel.size = bidSize;
-            curReturnParcel.targetingType = 'slot';
-            curReturnParcel.targeting = {};
+        var bidPrice = curBid.creatives[0].cpm;                                           /* the bid price for the given slot */
+        var bidSize = [1, 1];                                                             /* 1x1 because we're native */
```

Comment:
Is it possible to return the actual creative size and not set bidSize to 1x1?

Reply:
We use 1x1 as a placeholder size. The nature of us being native is that we render our ad markup onto the parent page DOM such that it inherits the styling of the parent page and the size is determined by the parent page DOM.  The method of how we do that is outlined in the previous comment referencing `sfp-set-targeting.js`